### PR TITLE
Reports Legend

### PR DESCRIPTION
**Pull request by willsb**
Ref #812

This moves most of the logic to inside a header in the ReportsViewController to make scrolling possible.

Squash ASAP

### CODE REVIEW

File: Toggl.Daneel/ViewSources/ReportsTableViewSource.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        }
+
+        protected override UITableViewCell GetOrCreateCellFor(UITableView tableView, NSIndexPath indexPath, object item)
+            => tableView.DequeueReusableCell(cellIdentifier, indexPath);
+
+        public override nfloat GetHeightForRow(UITableView tableView, NSIndexPath indexPath) => 56;
+
+        public override nfloat GetHeightForHeader(UITableView tableView, nint section) => 535;
+    }
+}
```
Comment (amulware) [inline Toggl.Daneel/ViewSources/ReportsTableViewSource.cs]: \n

File: Toggl.Foundation.Tests/Reports/ProjectSummaryReportTests.cs [hunk 2]
```diff
@@ -18,7 +18,7 @@ public void CalculatesThePercentageOfBillable(NonEmptyArray<NonNegativeInt> dura
                 var segments = durations.Get.Select(duration => duration.Get)
                     .Select((index, duration) => getSegmentFromDurationAndIndex(index, duration))
                     .ToArray();
-                var totalTrackedSeconds = segments.Select(s => s.TrackedSeconds).Sum();
+                var totalTrackedSeconds = segments.Select(s => s.TrackedTime).Sum();
```
Comment (amulware) [inline Toggl.Foundation.Tests/Reports/ProjectSummaryReportTests.cs]: Is there an extension to sum timespans? Or how does this work?

File: Toggl.Foundation.Tests/Reports/ProjectSummaryReportTests.cs [hunk 3]
```diff
@@ -18,7 +18,7 @@ public void CalculatesThePercentageOfBillable(NonEmptyArray<NonNegativeInt> dura
...
                 var billableSeconds = segments.Select(s => s.BillableSeconds).Sum();
                 var expectedBillablePercentage = totalTrackedSeconds > 0 ? (100.0f / totalTrackedSeconds) * billableSeconds : 0;
```
Comment (amulware) [inline Toggl.Foundation.Tests/Reports/ProjectSummaryReportTests.cs]: Dividing a number by a timespan?

File: Toggl.Foundation.Tests/Reports/ProjectSummaryReportTests.cs [hunk 4]
```diff
@@ -34,7 +34,7 @@ public void CalculatesTheTotalAmountOfSeconds(NonEmptyArray<NonNegativeInt> dura
                 var segments = actualDurations
                     .Select((index, duration) => getSegmentFromDurationAndIndex(index, duration))
                     .ToArray();
-                var expectedDuration = segments.Select(s => s.TrackedSeconds).Sum();
+                var expectedDuration = segments.Select(s => s.TrackedTime).Sum();
```
Comment (amulware) [inline Toggl.Foundation.Tests/Reports/ProjectSummaryReportTests.cs]: See above.

**Comment by amulware** [comment]
Oh, the answer to my questions is: It's not actually building right now. Needs fix.

File: Toggl.Foundation.Tests/Reports/ProjectSummaryReportTests.cs (hunk 2 above)
Comment (willsb) [reply Toggl.Foundation.Tests/Reports/ProjectSummaryReportTests.cs]: Tests are still broken, as mentioned on slack

File: Toggl.Foundation.Tests/Reports/ProjectSummaryReportTests.cs (hunk 3 above)
Comment (willsb) [reply Toggl.Foundation.Tests/Reports/ProjectSummaryReportTests.cs]: Tests are still broken, as mentioned on slack

**Comment by pe0ny** [comment]
reports

@willsb In simulator (iphone SE) this happens ☝️ 
I believe the correct behaviour should be this: Pie chart section with the legend scrolls to the top (together with pie chart) and when list items are more than fits to the screen, list items start scrolling and pie chart sticks to the top.

**Comment by pe0ny** [comment]
Also, maybe someone can confirm but for me, this view behaves rather differently on different screens:
- On SE the legend is not visible (see the gif above)
- On iPhone 6s pie chart is not visible
- On iPhone 6S Plus it works fine 👌

File: Toggl.Foundation/Reports/ReportsProvider.cs [hunk 5]
```diff
@@ -45,17 +45,26 @@ private IObservable<ProjectSummaryReport> summaryReportFromResponse(IProjectsSum
...
+            
+                    })
+                    .Select(segments => 
+                        new ProjectSummaryReport(segments.OrderByDescending(c => c.Percentage).ToArray()));
+
+        private static ChartSegment chartFromSummary(IProjectSummary summary, 
+                                                     IList<IProject> projectsInReport,
+                                                     long totalSeconds)
         {
+            var percentage = (summary.TrackedSeconds / totalSeconds) * 100;
```
Comment (simonrozsival) [inline Toggl.Foundation/Reports/ReportsProvider.cs]: Small long / big long == 0

**Comment by simonrozsival** [comment]
Functional test on an iPhone 8 in simulator 👌

**Pull request merged**